src/app/components/feed/jh-pushtotalk.directive.js

```diff
@@ -0,0 +1,148 @@
+        var lastHotkey = UserService.getSetting("lastHotkey");
+        if (lastHotkey) {
+          vm.hotkeyActive = true;
+          setPushToTalk(lastHotkey);
+        }
+      }
+
+      function click($event) {
+        if (ignoreClick) {
+          return;
+        }
+
+        $event.currentTarget.blur();
+
+        if (vm.hotkeyActive) {
+          if (vm.hotkey !== null) {
+            setPushToTalk(null);
+          } else {
+            window.Mousetrap.stopRecord();
+          }
+          vm.hotkeyActive = false;
+        } else {
+          vm.hotkeyActive = true;
+          recordSequence();
+        }
+      }
+
+      function recordSequence() {
+        vm.toggleText = "Choose a hotkey for Push-to-Talk...";
+
+        var recordCallback = function(sequence) {
+          vm.toggleText = "";
+
+          if (sequence !== undefined && sequence !== null && sequence[0] !== undefined && sequence[0].length > 0) {
+			  //We don't want to support key combinations because the keyup event is not fired when the keys are released in wrong order.
+			  // This problem needs to be fixed in angular hotkeys / Mousetrap library. Second problem is broken support for the right super key.
+              if (sequence[0].indexOf('+') > -1 && sequence[0].length !== 1 || sequence[0] === '\\') {
+                warn("Sorry, key combinations are not supported!");
+              } else {
+                setPushToTalk(sequence[0]);
+              }
+          } else {
+            vm.hotkeyActive = false;
+          }
+        };
+
+        window.Mousetrap.record({
+          recordSequence: false
+        }, function(sequence) {
+          recordCallback(sequence);
+        });
+      }
+
+      function warn(warning) {
+        vm.toggleText = warning;
+        ignoreClick = true;
+
+        var timeoutCallback = function() {
+          vm.toggleText = "";
+          vm.hotkeyActive = false;
+          ignoreClick = false;
+        };
+
+        $timeout(timeoutCallback, 3000);
+      }
+
+      function setPushToTalk(key) {
+        if (vm.hotkey !== null) {
+          hotkeys.del(vm.hotkey, 'keydown');
+          hotkeys.del(vm.hotkey, 'keyup');
+
+          vm.hotkey = null;
+          vm.showHotkey = false;
+          vm.titleText = titleText1;
+        }
+
+        if (key !== null) {
+          var pttCallback = function(event) {
+            event.preventDefault();
+            RoomService.pushToTalk(event.type);
+          };
+
+          hotkeys.add({
+            combo: key,
+            description: 'Push-to-talk',
+            callback: pttCallback,
+            action: 'keydown'
+          });
+          hotkeys.add({
+            combo: key,
+            description: 'Push-to-talk',
+            callback: pttCallback,
+            action: 'keyup'
+          });
+
+          vm.hotkey = key;
+          vm.showHotkey = true;
+          vm.titleText = titleText2;
+          UserService.setSetting("lastHotkey", key);
```

Comment:
This persists the key when push-to-talk is set. But it does not "forget" the setting when push-to-talk is disabled. That's unexpected. If I enable push-to-talk once, I will have it automatically enabled on every subsequent login, even if I disable it every time.